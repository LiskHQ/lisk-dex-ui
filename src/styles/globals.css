:root {
  font-size: 16px;
}

html,
body,
#__next {
  padding: 0 !important;
  margin: 0;
  height: 100%;
}

a {
  color: inherit;
  text-decoration: none;
}

* {
  box-sizing: border-box;
}

a,
.btn,
button,
span,
p,
i,
input,
select,
textarea,
li,
img,
svg path,
*::after,
*::before,
.transition-3,
h1,
h2,
h3,
h4,
h5,
h6 {
  transition: all 0.3s ease-out 0s;
}

span,
p,
a,
li.MuiMenuItem-root,
label,
label.MuiFormLabel-root,
span.MuiTypography-root,
p.MuiTypography-root,
button.MuiButton-root,
button.MuiButtonBase-root,
.MuiInput-root,
.MuiTableCell-root {
  font-family: 'Roboto', sans-serif;
}

h1,
h2,
h3,
h4,
h5,
h6,
h1.MuiTypography-root,
h2.MuiTypography-root,
h3.MuiTypography-root,
h4.MuiTypography-root,
h5.MuiTypography-root,
h6.MuiTypography-root {
  font-family: 'Roboto', sans-serif;
}