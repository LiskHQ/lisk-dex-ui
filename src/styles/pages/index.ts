import { GovernanceStyle } from "./governance/index.style";

export {
  GovernanceStyle
}