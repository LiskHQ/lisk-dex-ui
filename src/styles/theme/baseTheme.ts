export const baseTheme = {
  typography: {
    h1: {
      fontSize: "2.5rem",
      fontWeight: 600,
    },
    h2: {
      fontSize: "2rem",
      fontWeight: 600,
    },
    h3: {
      fontSize: "1.75rem",
      lineHeight: "2rem",
      fontWeight: 500,
    },
    h4: {
      fontSize: "1.25rem",
      fontWeight: 600,
    },
    h5: {
      fontSize: "1rem",
      lineHeight: "1.25rem",
      fontWeight: 500,
    },
    h6: {
      fontSize: "0.875rem",
      lineHeight: "1rem",
      fontWeight: 500,
    },
    body1: {
      fontSize: "1rem",
      lineHeight: "1.375rem",
      fontWeight: 500,
    },
    body2: {
      fontSize: "0.875rem",
      lineHeight: "1.125rem",
      fontWeight: 400,
    },
    caption: {
      fontSize: "0.875rem",
      lineHeight: "1rem",
      fontWeight: 500,
    },
    subtitle1: {
      fontSize: "1.75rem",
      lineHeight: "2rem",
      fontWeight: 500,
    },
    subtitle2: {
      fontSize: "1.25rem",
      lineHeight: "1.5rem",
      fontWeight: 400,
    }
  },
  icons: {
    sm: "1rem",
    md: "1.25rem",
    lg: "1.5rem",
  },
  breakpoints: {
    values: {
      xs: 0,
      sm: 576,
      md: 768,
      lg: 1024,
      xl: 1440,
    },
  },
  shadows: {
    sm: "0px 1px 3px rgba(0, 0, 0, 0.1)"
  }
}