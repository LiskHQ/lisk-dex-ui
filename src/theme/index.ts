export { default as theme } from "./theme";
export { default as variables } from "./variables";
