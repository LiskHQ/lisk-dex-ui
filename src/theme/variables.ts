const variables = {
  black: "#000000",
  white: "#ffffff",
  primary: {
    superPurple: "#6953F4",
    lighter: "#5845CC",
    light: "#4738A6",
    main: "#362B7F",
    dark: "#261E59",
    superDark: "#161133",
    contrastText: "#fff",
    stone: "#323d47",
    darkSilver: "#757575",
  },
  secondary: {
    superPurple: "#F5981B",
    lighter: "#CC7E16",
    light: "#A66712",
    main: "#7F4F0E",
    dark: "#59370A",
    superDark: "#332006",
  },
  opacities: {
    superPurple: "#F5981B",
    lighter: "#CC7E16",
    light: "#A66712",
    main: "#7F4F0E",
    dark: "#59370A",
    superDark: "#332006",
    tokenBackground: "#F2F5F9",
  },
  success: {
    superPurple: "#E3F2E0",
    lighter: "#8DC881",
    light: "#459A33",
  },
  warning: {
    superPurple: "#F8F3D8",
    lighter: "#CC7E16",
    light: "#C9A81C",
  },
  information: {
    superPurple: "#F5981B",
    lighter: "#CC7E16",
    light: "#A66712",
  },
  error: {
    superPurple: "#F5981B",
    lighter: "#CC7E16",
    light: "#A66712",
  },
  background: "#E1E3E6",
  lineColor: "#e0e2f1",
};

export default variables;
