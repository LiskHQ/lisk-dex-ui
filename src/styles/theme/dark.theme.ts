import { createTheme } from '@mui/material/styles';
import { baseTheme } from './baseTheme';

const themeColors = {
  text: {
    primary: '#F2F2F2',
    secondary: '#F2F2F2',
    paragraph: '#CDCDCD',
    body: '#F2F2F2',
    heading: '#4738A6',
    button: '#FFFFFF',
    placeholder: '#A4ABB7',
  },
  lightcurve: {
    0: '#6953F4',
    10: '#5845CC',
    20: '#4738A6',
    40: '#362B7F',
    60: '#261E59',
    80: '#E6E6E6',
  },
  secondary: {
    0: '#F5981B'
  },
  primary: {
    85: '#7E6CF4',
    70: '#4F39B2',
    60: '#271B5B',
    40: '#2D2166',
    20: '#322573',
    10: '#4F39B2',
    5: '#322573',
    2.5: '#25195E',
    1: '#F4F8F9',
    0: '#FFFFFF',
  },
  opacities: {
    5: '#333333',
    10: '#3D3D3D',
    20: '#E6E6E6',
    40: '#7E7E7E',
    80: '#D4D4D4',
  },
  border: {
    primary: '#433199',
  },
  bg: {
    primary: 'linear-gradient(90deg, #140C42 0%, #211953 52.88%, #110A3A 101.53%, #140C42 101.53%)',
    secondary: '#22184C',
    helper: '#6953F4',
    proposalItem: 'linear-gradient(180deg, #2E2567 0%, rgba(37, 26, 103, 0.5) 143.01%);',
    banner: 'linear-gradient(180deg, #150556 0%, rgba(25, 9, 98, 0.9) 100%)',
    walletAddress: 'linear-gradient(180deg, #372589 0%, rgba(27, 19, 68, 0.8) 100%)',
    slider: '#322573',
  },
  success: {
    primary: '#E3F2E0',
    second: '#8DC881',
  },
  warning: {
    primary: '#F8F3D8',
    second: '#E2CA64',
  },
  info: {
    primary: '#2EA3DD',
    second: '#C0ECE8',
  },
  error: {
    primary: '#F6E4E6',
    second: '#FB8B96',
  },
} as const;

const lightTheme = createTheme({
  ...themeColors,
  ...baseTheme,
  palette: {
    mode: 'dark',
  },
})

export default lightTheme
