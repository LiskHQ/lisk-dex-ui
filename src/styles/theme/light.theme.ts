import { createTheme } from '@mui/material/styles';
import { baseTheme } from './baseTheme';

const themeColors = {
  text: {
    primary: '#282828',
    secondary: '#515661',
    paragraph: '#6B7280',
    body: '#6B7280',
    heading: '#4738A6',
    button: '#FFFFFF',
    placeholder: '#6B7280',
  },
  lightcurve: {
    0: '#6953F4',
    10: '#A196F6',
    20: '#4738A6',
    40: '#362B7F',
    60: '#261E59',
    80: '#161133',
  },
  primary: {
    85: '#7E6CF4',
    60: '#A196F6',
    40: '#E4EBF9',
    20: '#D9D8F8',
    10: '#E7E9F9',
    5: '#EEF0F8',
    2.5: '#F2F5F9',
    1: '#F4F8F9',
    0: '#6953F4'
  },
  opacities: {
    5: '#333333',
    10: '#3D3D3D',
    20: '#535353',
    40: '#7E7E7E',
    80: '#D4D4D4',
  },
  border: {
    primary: '#E6E6E6',
  },
  bg: {
    primary: '#FBFBFB',
    secondary: '#FFFFFF',
    helper: '#22184D',
    banner: 'linear-gradient(180deg, #F3F3FD 0%, rgba(238, 238, 249, 0.9) 100%)',
    walletAddress: 'linear-gradient(180deg, #EEEFF7 0%, rgba(232, 234, 245, 0.8) 100%);',
    slider: '#F2F2F2',
  },
  success: {
    primary: '#E3F2E0',
    second: '#8DC881',
  },
  warning: {
    primary: '#F8F3D8',
    second: '#E2CA64',
  },
  info: {
    primary: '#EBF7F6',
    second: '#C0ECE8',
  },
  error: {
    primary: '#F6E4E6',
    second: '#FB8B96',
  },
} as const;

const lightTheme = createTheme({
  ...themeColors,
  ...baseTheme,
  palette: {
    mode: 'light'
  },
})

export default lightTheme
