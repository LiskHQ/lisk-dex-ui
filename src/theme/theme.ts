import {
  createTheme,
  responsiveFontSizes,
  ThemeOptions,
} from "@mui/material/styles";

import variables from "./variables";

const options: ThemeOptions = {
  palette: {
    primary: variables.primary,
    secondary: variables.secondary,
  },
  typography: {
    h1: {
      fontSize: "2.5rem",
      fontWeight: 600,
    },
    h2: {
      fontSize: "2rem",
      fontWeight: 400,
    },
    h3: {
      fontSize: "1.75rem",
      fontWeight: 400,
    },
    h4: {
      fontSize: "1.25rem",
      fontWeight: 400,
    },
    h5: {
      fontSize: "1rem",
      fontWeight: 400,
    },
    h6: {
      fontSize: "0.75rem",
      fontWeight: 400,
    },
  },
  breakpoints: {
    values: {
      xs: 0,
      sm: 576,
      md: 768,
      lg: 1024,
      xl: 1440,
      xxl: 1950,
      xxxl: 2300,
    },
  },
};

const theme = responsiveFontSizes(createTheme(options));
export default theme;
