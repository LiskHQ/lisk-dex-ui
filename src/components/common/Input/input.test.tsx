import { ThemeProvider } from '@mui/material';
import { render } from '@testing-library/react';
import React from 'react';
import { lightTheme } from 'styles/theme';
import { InputComponent } from '.';

function renderComponent() {
  return render(
    <ThemeProvider theme={lightTheme}>
      <InputComponent />
    </ThemeProvider>
  );
}

describe('Input component', () => {
  it('checks if the component matches the snapshot', () => {
    const { container } = renderComponent();
    expect(container).toMatchSnapshot();
  });
});