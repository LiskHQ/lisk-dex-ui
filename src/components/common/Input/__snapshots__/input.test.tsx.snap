// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input component checks if the component matches the snapshot 1`] = `
.emotion-0 {
  position: relative;
}

.emotion-0 .MuiInputLabel-root {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  position: unset;
  color: #282828!important;
  font-size: 0.875rem;
  line-height: 1.124rem;
  font-weight: 400;
  -webkit-transform: none;
  -moz-transform: none;
  -ms-transform: none;
  transform: none;
  -webkit-transform: none;
  -ms-transform: none;
}

.emotion-0 .MuiInputLabel-root svg {
  cursor: pointer;
  margin-left: 0.25rem;
}

.emotion-0 .MuiInputLabel-root svg path {
  stroke: #6B7280!important;
}

.emotion-0 .MuiInputBase-root {
  padding: 0.75rem 1rem;
  border-radius: 0.25rem;
  color: #282828;
  border: 0.5px solid #E6E6E6;
  background: unset;
  -webkit-animation-duration: 0.5s!important;
  animation-duration: 0.5s!important;
  margin-top: 0.5rem;
  font-size: 1rem;
  line-height: 1.625rem;
}

.emotion-0 .MuiInputBase-root input {
  padding: 0;
}

.emotion-0 .MuiInputBase-root fieldset {
  border: none;
}

.emotion-0 .MuiInputBase-root:before {
  content: none;
}

.emotion-0 .MuiInputBase-root:after {
  content: none;
}

.emotion-0 .MuiInputBase-root:hover {
  border: 1px solid #5845CC;
}

.emotion-0 .MuiInputBase-root:focus-within {
  border: 1px solid #6953F4;
}

.emotion-0 .MuiFormHelperText-root {
  color: #FB8B96;
  margin: 0.5rem 0 0 0;
}

.emotion-1 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  position: relative;
  min-width: 0;
  padding: 0;
  margin: 0;
  border: 0;
  vertical-align: top;
}

.emotion-2 {
  font-size: 1rem;
  line-height: 1.4375em;
  font-weight: 500;
  font-family: "Roboto","Helvetica","Arial",sans-serif;
  letter-spacing: 0.00938em;
  color: rgba(0, 0, 0, 0.87);
  box-sizing: border-box;
  position: relative;
  cursor: text;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  position: relative;
  border-radius: 4px;
}

.emotion-2.Mui-disabled {
  color: rgba(0, 0, 0, 0.38);
  cursor: default;
}

.emotion-2:hover .MuiOutlinedInput-notchedOutline {
  border-color: rgba(0, 0, 0, 0.87);
}

@media (hover: none) {
  .emotion-2:hover .MuiOutlinedInput-notchedOutline {
    border-color: rgba(0, 0, 0, 0.23);
  }
}

.emotion-2.Mui-focused .MuiOutlinedInput-notchedOutline {
  border-color: #1976d2;
  border-width: 2px;
}

.emotion-2.Mui-error .MuiOutlinedInput-notchedOutline {
  border-color: #d32f2f;
}

.emotion-2.Mui-disabled .MuiOutlinedInput-notchedOutline {
  border-color: rgba(0, 0, 0, 0.26);
}

.emotion-3 {
  font: inherit;
  letter-spacing: inherit;
  color: currentColor;
  padding: 4px 0 5px;
  border: 0;
  box-sizing: content-box;
  background: none;
  height: 1.4375em;
  margin: 0;
  -webkit-tap-highlight-color: transparent;
  display: block;
  min-width: 0;
  width: 100%;
  -webkit-animation-name: mui-auto-fill-cancel;
  animation-name: mui-auto-fill-cancel;
  -webkit-animation-duration: 10ms;
  animation-duration: 10ms;
  padding: 16.5px 14px;
}

.emotion-3::-webkit-input-placeholder {
  color: currentColor;
  opacity: 0.42;
  -webkit-transition: opacity 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  transition: opacity 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
}

.emotion-3::-moz-placeholder {
  color: currentColor;
  opacity: 0.42;
  -webkit-transition: opacity 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  transition: opacity 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
}

.emotion-3:-ms-input-placeholder {
  color: currentColor;
  opacity: 0.42;
  -webkit-transition: opacity 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  transition: opacity 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
}

.emotion-3::-ms-input-placeholder {
  color: currentColor;
  opacity: 0.42;
  -webkit-transition: opacity 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  transition: opacity 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
}

.emotion-3:focus {
  outline: 0;
}

.emotion-3:invalid {
  box-shadow: none;
}

.emotion-3::-webkit-search-decoration {
  -webkit-appearance: none;
}

label[data-shrink=false]+.MuiInputBase-formControl .emotion-3::-webkit-input-placeholder {
  opacity: 0!important;
}

label[data-shrink=false]+.MuiInputBase-formControl .emotion-3::-moz-placeholder {
  opacity: 0!important;
}

label[data-shrink=false]+.MuiInputBase-formControl .emotion-3:-ms-input-placeholder {
  opacity: 0!important;
}

label[data-shrink=false]+.MuiInputBase-formControl .emotion-3::-ms-input-placeholder {
  opacity: 0!important;
}

label[data-shrink=false]+.MuiInputBase-formControl .emotion-3:focus::-webkit-input-placeholder {
  opacity: 0.42;
}

label[data-shrink=false]+.MuiInputBase-formControl .emotion-3:focus::-moz-placeholder {
  opacity: 0.42;
}

label[data-shrink=false]+.MuiInputBase-formControl .emotion-3:focus:-ms-input-placeholder {
  opacity: 0.42;
}

label[data-shrink=false]+.MuiInputBase-formControl .emotion-3:focus::-ms-input-placeholder {
  opacity: 0.42;
}

.emotion-3.Mui-disabled {
  opacity: 1;
  -webkit-text-fill-color: rgba(0, 0, 0, 0.38);
}

.emotion-3:-webkit-autofill {
  -webkit-animation-duration: 5000s;
  animation-duration: 5000s;
  -webkit-animation-name: mui-auto-fill;
  animation-name: mui-auto-fill;
}

.emotion-3::-webkit-input-placeholder {
  opacity: 1;
  color: #6B7280;
}

.emotion-3::-moz-placeholder {
  opacity: 1;
  color: #6B7280;
}

.emotion-3:-ms-input-placeholder {
  opacity: 1;
  color: #6B7280;
}

.emotion-3::placeholder {
  opacity: 1;
  color: #6B7280;
}

.emotion-3::-webkit-outer-spin-button,
.emotion-3::-webkit-inner-spin-button {
  -webkit-appearance: none;
  display: none;
}

.emotion-3:-webkit-autofill {
  border-radius: inherit;
}

.emotion-3::-webkit-input-placeholder {
  opacity: 1;
  color: #6B7280;
}

.emotion-3::-moz-placeholder {
  opacity: 1;
  color: #6B7280;
}

.emotion-3:-ms-input-placeholder {
  opacity: 1;
  color: #6B7280;
}

.emotion-3::placeholder {
  opacity: 1;
  color: #6B7280;
}

.emotion-3::-webkit-outer-spin-button,
.emotion-3::-webkit-inner-spin-button {
  -webkit-appearance: none;
  display: none;
}

.emotion-4 {
  text-align: left;
  position: absolute;
  bottom: 0;
  right: 0;
  top: -5px;
  left: 0;
  margin: 0;
  padding: 0 8px;
  pointer-events: none;
  border-radius: inherit;
  border-style: solid;
  border-width: 1px;
  overflow: hidden;
  min-width: 0%;
  border-color: rgba(0, 0, 0, 0.23);
}

.emotion-5 {
  float: unset;
  overflow: hidden;
  padding: 0;
  line-height: 11px;
  -webkit-transition: width 150ms cubic-bezier(0.0, 0, 0.2, 1) 0ms;
  transition: width 150ms cubic-bezier(0.0, 0, 0.2, 1) 0ms;
}

<div>
  <div
    class="MuiBox-root emotion-0"
  >
    <div
      class="MuiFormControl-root MuiTextField-root emotion-1"
    >
      <div
        class="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-formControl emotion-2"
      >
        <input
          aria-invalid="false"
          class="MuiOutlinedInput-input MuiInputBase-input emotion-3"
          id=":r0:"
          type="text"
          value=""
        />
        <fieldset
          aria-hidden="true"
          class="MuiOutlinedInput-notchedOutline emotion-4"
        >
          <legend
            class="emotion-5"
          >
            <span
              class="notranslate"
            >
              â€‹
            </span>
          </legend>
        </fieldset>
      </div>
    </div>
  </div>
</div>
`;
